### PR TITLE
For #10162 - Don't show the contextual menu for blocked URLs

### PR DESCRIPTION
**Pull request by Mugurell**
There is no point in showing the "Open in new tab", "Save.." contextual menu
options and all the others if that url is blocked from loading.

Testing on https://jsfiddle.net/14sd6hyj/show

| Before | After |
| -- | -- |
| <video src="https://user-images.githubusercontent.com/11428869/116260110-1500ef00-a77f-11eb-8d6d-550af0a08125.mp4" /> | <video src="https://user-images.githubusercontent.com/11428869/116260166-1f22ed80-a77f-11eb-9f37-7fc770714d67.mp4" /> |

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Mugurell** [comment]
```
unhandled URLError; reason type: str; value: Internal Server ErrorPERFHERDER_DATA: {"framework": {"name": "vcs"}, "suites": [{"extraOptions": ["m5d.large"], "lowerIsBetter": true, "name": "overall", "serverUrl": "hg.mozilla.org", "shouldAlert": false, "subtests": [], "value": 105.79771399497986}, {"extraOptions": ["m5d.large"], "lowerIsBetter": true, "name": "overall_nopull", "serverUrl": "hg.mozilla.org", "shouldAlert": false, "subtests": [], "value": 105.79771399497986}, {"extraOptions": ["m5d.large"], "lowerIsBetter": true, "name": "overall_nopull_fullcheckout", "serverUrl": "hg.mozilla.org", "shouldAlert": false, "subtests": [], "value": 105.79771399497986}, {"extraOptions": ["m5d.large"], "lowerIsBetter": true, "name": "overall_nopull_populatedwdir", "serverUrl": "hg.mozilla.org", "shouldAlert": false, "subtests": [], "value": 105.79771399497986}]}
abort: HTTP Error 500: Internal Server Error
```

Restarting CI.

File: components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt [hunk 1]
```diff
@@ -1029,6 +1033,52 @@ class ContextMenuCandidateTest {
...
+    @Test
+    fun `GIVEN SessionState with blocked url schemes WHEN isUrlSchemeAllowed is called THEN it returns false if the url has that scheme`() {
+        val engineSessionWithBlockedUrlScheme = Mockito.mock(EngineSession::class.java)
+        doReturn(listOf("http")).`when`(engineSessionWithBlockedUrlScheme).getBlockedSchemes()
+        val sessionState = TabSessionState(
+            content = mock(),
+            engineState = EngineState(engineSession = engineSessionWithBlockedUrlScheme)
+        )
+
+        assertFalse(sessionState.isUrlSchemeAllowed("http://mozilla.org"))
```
Comment (csadilek) [inline components/feature/contextmenu/src/test/java/mozilla/components/feature/contextmenu/ContextMenuCandidateTest.kt:1068]: Can we add a few more edge cases here e.g. `hTtPs://` (which I know will work because you used `normalizedScheme` but good to guard against) and maybe a url without a protocol `mozilla.org` or `/path`.

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt [hunk 2]
```diff
@@ -660,4 +660,9 @@ abstract class EngineSession(
      */
     @CallSuper
     open fun close() = delegate.unregisterObservers()
+
+    /**
+     * List of URL schemes that are blocked from loading.
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/EngineSession.kt:665]: nit: Returns the list of ...

**Review by csadilek — APPROVED** [review]
Looks good to me! A nit and a suggestion below to cover more cases in the tests.
